cr.c

```diff
@@ -84,9 +84,11 @@ static void dill_ctx_timer_handler(int sig, siginfo_t *si, void *uc) {
 #ifndef __APPLE__
     struct dill_ctx_cr *ctx = si->si_value.sival_ptr;
     int or = timer_getoverrun(ctx->timer);
-    if(or) ctx->do_poll = 1;
+    if(or) {
+        __sync_lock_test_and_set(&ctx->do_poll, 1);
+    }
 #else
-    dill_poll_count++;
+    __sync_fetch_and_add(&dill_poll_count, 1);
```

Comment:
What about doing the following optimization: now() does rdtsc, compares it with stored tsc from the last time clock_gettime() was done. It it's less than N, it doesn't even bother going clock_gettime(), instead it returns stored value from the last time it was called. If it's larger than N it does clock_gettime(), caches that value and the tsc.

The effect should be that now() has basically the cost of rdtsc, except for once a millisecond when full clock_gettime() is done.